### PR TITLE
Add Github CodeQL security checks

### PR DESCRIPTION
**Pull request by jfcg**
Add Github CodeQL security checks for C++, Python.

In my fork under `/security/code-scanning` I see:

godot

### CODE REVIEW

**Comment by pycbouh** [comment]
Please squash your commits into one, and use a descriptive name for that one commit. For example, your PR title is fine.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+
+    - run: |
+       sudo apt-get install build-essential scons pkg-config libx11-dev libxcursor-dev libxinerama-dev libgl1-mesa-dev libglu-dev libasound2-dev libpulse-dev libxi-dev libxrandr-dev yasm
+       scons platform=x11 target=release_debug udev=no -j 4
```
Comment (qarmin) [inline .github/workflows/codeql-analysis.yml:31]: Each command should have different steps:
- Dependencies
- Compilation
- Run

etc. just like in others CI files

**Review by qarmin — CHANGES_REQUESTED** [review]

**Comment by qarmin** [comment]
This PR doesn't contains any caching mechanism(look at other CI jobs for more info), so each time it will need to compile ~50 minutes entire engine

**Comment by qarmin** [comment]
> In my fork under `/security/code-scanning` I see:

This is hidden for me and probably for other people also.

**Comment by Calinou** [comment]
> This PR doesn't contains any caching mechanism(look at other CI jobs for more info), so each time it will need to compile ~50 minutes entire engine

Due to the long build times, it may be better to not bother with building pull requests and only build pushes to the upstream repository.

**Comment by qarmin** [comment]
Only 25 from 304 reports comes from Godot and the rest from thirdparty repositories.
This tool only found in code `Multiplication result converted to larger type`.

I was expecting a Cppcheck-like error detector, and with such results I don't think that `Github CodeQL` is good choice.

**Comment by jfcg** [comment]
3rd party libraries are part of your codebase, and they may not be necessarily scanned for security checks, i dont know about the overall false positive rate, a deeper analysis of the alerts could be necessary.

**Comment by aaronfranke** [comment]
Ideally, we can find and fix the bugs in third-party libraries, and contribute the fixes back upstream.

**Comment by akien-mga** [comment]
This seems interesting, it would be good to revisit it.

I'm not sure it's a good idea to run it on each PR as in my experience this kind of tool finds lots of false positives and it will take a long time to fix or work around all of them before we can make its failures blocking for a given PR. Maybe it could run once per day instead.

OP's fork has been deleted so we can't check how useful the results were.

Would be good to salvage if a contributor is interested.

**Pull request closed without merging**